source/fundamentals/aggregation.txt

```diff
@@ -0,0 +1,192 @@
+the `MongoCollection.aggregate() <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/MongoCollection.html#aggregate(com.mongodb.client.ClientSession,java.util.List,java.lang.Class)>`__
+method.
+
+The Java driver provides the `Aggregates <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html>`__
+helper class that
+contains builders for aggregation stages.
+
+In the following example, the aggregation pipeline:
+
+- Uses a :manual:`$match <reference/operator/aggregation/match/>` stage to filter for documents whose
+  ``categories`` array field contains the element ``Bakery``. The example uses
+  `Aggregates.match <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html#match(org.bson.conversions.Bson)>`__
+  to build the ``$match`` stage.
+
+- Uses  a :manual:`$group <reference/operator/aggregation/group/>` stage to group the matching documents by the ``stars``
+  field, accumulating a count of documents for each distinct value of ``stars``. The example uses
+  `Aggregates.group <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html#group(org.bson.conversions.Bson)>`__
+  to build the ``$group`` stage and `Accumulators.sum <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Accumulators#sum(java.lang.String,TExpression)>`__
+  to build the accumulator expression. The :manual:`accumulator expressions <reference/operator/aggregation/#accumulators-group>`
+  used within the ``$group`` stage can be built with the `Accumulators <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Accumulators.html>`__
+  helper class.
+
+.. code-block:: java
+
+   collection.aggregate(
+         Arrays.asList(
+                 Aggregates.match(Filters.eq("categories", "Bakery")),
+                 Aggregates.group("$stars", Accumulators.sum("count", 1))
+         )
+   ).forEach(printBlock);
+
+Use Aggregation Expressions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Java driver provides builders for :java-sync-api:`accumulator expressions<com/mongodb/client/model/Accumulators.html>`
+for use with :java-sync-api:`$group <com/mongodb/client/model/Aggregates.html#group(TExpression,java.util.List)>`. All
+other expressions must be manually built.
+
+In the following example, the aggregation pipeline uses a
+`$project <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html#project(org.bson.conversions.Bson)>`__
+stage and various
+`Projections <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Projections.html>`__
+to return only the ``name`` field, and the calculated field ``firstCategory`` whose value is the first element in the
```

Comment:
Would it be worthwhile linking to the "arrayElemAt" aggregation operator (in an API links section at the end)? Also, I wonder if it would be helpful to explain the "$" field path for categories.